### PR TITLE
Reworks the minting/coin press to be buildable and upgradable

### PR DESCRIPTION
**Pull request by senjx**

### CODE REVIEW

**Comment by senjx** [comment]
I didn't mean to make a pull request but I might as well keep it here. Do not merge for the meantime please, still WIP

**Comment by improvedname** [comment]
@kriskog verified tag please :^)

**Comment by Qustinnus** [comment]
WIP prs are not allowed it needs to be testmerge ready

File: code/modules/mining/mint.dm [hunk 1]
```diff
@@ -122,6 +122,19 @@
 				if (params["material_name"] == mat.name)
 					chosen = mat
 
+obj/machinery/mineral/mint/RefreshParts()
+
+	for(var/obj/item/stock_parts/matter_bin/B in component_parts) // Better matter bins increases the output of coins per cycle.
+	
+	for(var/obj/item/stock_parts/micro_laser/L in component_parts) // Better microlasers increase reduce the delay between production cycles.
+	
+	for(var/obj/item/stock_parts/manipulator/M in component_parts) // Better manipulators increase amount of coins produced per sheet.
```
Comment (Qustinnus) [inline code/modules/mining/mint.dm]: ah yes I too love infinite material duping

**Review by ghost — COMMENTED** [review]
coins are 500 of a material iirc, you can't make coin mints make more coinsxper sheet because then you're creating mats out of tin air which can then be recycled to be made into coins again.

**Comment by cacogen** [comment]
aye but at what cost?

File: code/modules/mining/mint.dm (hunk 1 above)
Comment (Dennok) [reply code/modules/mining/mint.dm]: Just make t4 100% effective.
And t1 25% effective.

Comment (Qustinnus) [reply code/modules/mining/mint.dm]: That'd be a good solution

**Comment by improvedname** [comment]
Tbh honestly dont add coin mint to circuit imprinter
Because i feel like that end in either somoene stealing a board from either engineering/cargo and print money to fund theif ID with and since you can eject all mats from a protolathe its gonna be counterproductive tbh.
Just give the QM a spare board

**Comment by cacogen** [comment]
can you use the existing coin press to become extremely rich and buy a bunch of prohibited shit through cargo at the cost of the station's material supply or what

**Comment by improvedname** [comment]
p.much

**Comment by trollbreeder** [comment]
stonks

**Comment by cacogen** [comment]
so much for time-gated money

File: code/modules/mining/mint.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mining/mint.dm]: check the code, it doesn't do that 😏

Comment (Qustinnus) [reply code/modules/mining/mint.dm]: what?

Comment (Qustinnus) [reply code/modules/mining/mint.dm]: oh yeah i realize but the comment shows they want to add it

**Comment by ExcessiveUseOfCobblestone** [comment]
just for reference, the coin vendor gives a different (higher) price for the amount of material used? If so that will need to be fixed for me to support this.

**Comment by cacogen** [comment]
i remember reading somewhere that @optimumtact prevented selling excess power to centcom because he wanted the amount of money in existence to be predictable based on how much time had passed (i.e. sourced mostly from paychecks). doesn't selling materials or minting coins go against that? i'd like money to actually have value instead of having all that excess lying around in people's accounts

**Comment by optimumtact** [comment]
as long as coins have a value in the currency and aren't just memorabilia, this can't fly.

**Comment by Dennok** [comment]
> as long as coins have a value in the currency and aren't just memorabilia, this can't fly.

You can sell mats for money, why you cant make coins fro mats and sell it for price of used mats?

**Comment by cacogen** [comment]
not to mention the fact that the coin press is already in play on lavaland

**Comment by optimumtact** [comment]
because those are mistakes, and need to be rectified, but they don't justify making more.

**Pull request closed without merging**